src/main/java/com/cegeka/academy/service/challenge/ChallengeServiceImp.java

```diff
@@ -69,6 +71,23 @@ public void saveChallenge(ChallengeDTO challengeDTO) {
         return userChallengesSet;
     }
 
+    @Override
+    public List<ChallengeDTO> getChallengesByCreatorId(Long creatorId) throws NotFoundException {
+
+        List<Challenge> challenges = challengeRepository.findAllByCreatorId(creatorId);
+
+        if(challenges == null || challenges.isEmpty()){
+
+            throw new NotFoundException().setMessage("List is empty.");
+        }
+
+        List<ChallengeDTO> challengeDTOList = challenges.stream().map(challenge -> ChallengeMapper.convertChallengeToChallengeDTO(challenge)).collect(Collectors.toList());
```

Comment:
aici poti pune direct ruturn, variabila challengeDTOList nu mai este folosita in alta parte.